pyensign/sync.py

```diff
@@ -0,0 +1,113 @@
+import asyncio
+import inspect
+
+
+def sync_to_async(fn):
+    """
+    Function that wraps a synchronous function inside of an async function and returns
+    the async function. Can be used as a function or a decorator.
+    """
+
+    if inspect.isgeneratorfunction(fn):
+        return gen_to_async(fn)
+    elif inspect.isfunction(fn):
+        return fn_to_async(fn)
+    else:
+        raise TypeError("expected a function or generator function")
+
+
+def async_to_sync(coro):
+    """
+    Function that wraps an async function inside of a synchronous function and returns
+    the synchronous function. Can be used as a function or a decorator.
+    """
+
+    if inspect.isasyncgenfunction(coro):
+        return async_gen_to_sync(coro)
+    elif inspect.iscoroutinefunction(coro):
+        return coro_to_sync(coro)
+    else:
+        raise TypeError("expected an async function or async generator function")
+
+
+def coro_to_sync(coro):
+    def wrap(*args, **kwargs):
+        # Ensure there is no running event loop
```

Comment:
Why should there be no running event loop? Can the coroutine not be put on the loop and awaited in the wrapper?

Reply:
I think it can be, I just learned about `asyncio.run_coroutine_threadsafe()` which takes an event loop as a parameter so that's how I'm handling this now.